libazureinit/src/provision/user.rs

```diff
@@ -96,26 +96,103 @@ impl Provisioner {
     }
 }
 
+#[instrument(skip_all)]
+fn user_exists(username: &str) -> Result<bool, Error> {
+    let status = Command::new("getent")
+        .arg("passwd")
+        .arg(username)
+        .status()?;
+
+    Ok(status.success())
+}
+
 #[instrument(skip_all)]
 fn useradd(user: &User) -> Result<(), Error> {
+    if user_exists(&user.name)? {
+        tracing::info!(
+            "User '{}' already exists. Skipping user creation.",
+            user.name
+        );
+
+        let current_groups_output =
```

Comment:
This is not necessary. You can just apply usermod -aG. usermod will add the user to the specified groups without affecting existing membership (and won't give an error if the user is already part of any of the groups)